mobgap/sl/_stride_length.py

```diff
@@ -0,0 +1,296 @@
+    5. Integration of vertical speed --> vertical displacement d(t)
+    6. Compute total vertical displacement during the step (d_step):
+        d_step = |max(d(t)) - min(d(t))|
+    7. Biomechanical model:
+        StepLength = A * 2 * sqrt(2 * LBh * d_step - d_step^2)
+        A: tuning coefficient, optimized by training for each population.
+        LBh: sensor height in meters, representative of the height of the center of mass.
+    8. Remove step length outliers during the gait sequence --> Hampel filter based on median absolute deviation.
+    9. Interpolation of StepLength values --> Step length per second values.
+    10. Remove length per second outliers during the gait sequence --> Hampel filter based on median absolute deviation.
+    11. Approximated stride length per second values = 2* step length per second values
+
+
+    This is based on the implementation published as part of the mobilised project [3]_.
+    However, this implementation deviates from the original implementation in some places.
+    For details, see the notes section and the examples.
+
+    %(sl_short)s
+
+    Parameters
+    ----------
+    %(sl_common_paras)s
+
+    Other Parameters
+    ----------------
+    %(other_parameters)s
+
+    Attributes
+    ----------
+    %(sl_sec_list_)s
+    %(stpl_list_)s
+
+    Notes
+    -----
+    %(sl_notes)s
+
+    -
+
+    .. [1] W. Zijlstra, & A. L. Hof, "Assessment of spatio-temporal gait parameters from trunk accelerations during
+        human walking" Gait & posture, vol. 18, no. 2, pp. 1-10, 2003.
+    .. [2] A. Soltani, et al. "Algorithms for walking speed estimation using a lower-back-worn inertial sensor:
+        A cross-validation on speed ranges." IEEE TNSRE, vol 29, pp. 1955-1964, 2021.
+    .. [3] https://github.com/mobilise-d/Mobilise-D-TVS-Recommended-Algorithms/blob/master/SLA_SLB/STRIDELEN.m
+
+    """
+
+    orientation_method: MadgwickAHRS
+    step_length_smoothing: BaseFilter
+    max_interpolation_gap_s: int
+    class PredefinedParameters:
+        """
+        The tuning coefficients to be used in the biomechanical model.
+        The reported parameters were previously accessed with model.K.ziljsV3.
+        Coefficients are divided by four to separate the coefficient from
+        the conversion from step length to stride length and from the x2 factor of the formula declared
+        in [1, 2].
+        """
+        tuning_coefficient_MS_MS = {"tuning_coefficient": 4.587/4, "tuning_cohort": "MS_MS"}
+        tuning_coefficient_MS_ALL = {"tuning_coefficient": 4.739/4, "tuning_cohort": "MS_ALL"}
+    @set_defaults(**PredefinedParameters.tuning_coefficient_MS_MS)
+    def __init__(
+        self,
+        *,
+        orientation_method: MadgwickAHRS = None,
+        step_length_smoothing: BaseFilter = cf(HampelFilter(2, 3.0)),
+        max_interpolation_gap_s: int = 3,
+        tuning_coefficient: float,
+        tuning_cohort: str,
```

Comment:
Do we need `tuning_cohort`? Is this used for something?

Reply:
Actually no, I thought it could be useful to hardcode it in case that someone wants to know the cohorts that the values correspond to. but I guess that it can be documented somewhere else (also, it is explicitated in the name of the tuning_coefficient e.g., tuning_coefficient_MS_MS)